### ISSUE TITLE
Controller tank heater operation: will it be possible to start/stop the tank heater and control the temperature?

### ISSUE THREAD

**Issue by McTrex** [speleolontra/daikin_residential_altherma#156]
Hi,

Not really an issue, but more of a feature request.
The daikin altherma integration (https://github.com/tadasdanielius/daikin_altherma/tree/main), has these options:

For water heating:

- Get/Set target temperature
- Get water temperature
- Turn off/on
- Turn powerful

Any chance they will be available in this integration in a future release?

Kind regards,
Marco

**Comment by arn1s-m** [comment]
Already there:
![image](https://github.com/speleolontra/daikin_residential_altherma/assets/58973493/69c2491c-3609-42dd-b74f-a100997e20cf)
![image](https://github.com/speleolontra/daikin_residential_altherma/assets/58973493/546f2873-24b2-423b-8cd1-1de556ce4b75)

**Comment by jwillemsen** [comment]
I do have that support with my altherma and this integration. Check the altherma 3 water heater entity

**Comment by McTrex** [comment]
Yes, I have that screen too, but if I'm not mistaken, that is for the house heating, not for the tapwater tank, right?
I have two tanks, one for heating, the other for hot tap water. I want to control the latter one.

**Comment by arn1s-m** [comment]
My first screen is for water heater - see currently at 50 degrees C. Options there are "Performance" (powerful in Daikin app), off and "Heat pump" (normal 'state').

2nd screen is for water heater for my house heating - currently off - otherwise I can adjust offset from the current temp

**Comment by jwillemsen** [comment]
When you go to the Settings/Devices&Integrations/Daikin Residential Controller for Altherma, click the "x entities", to show all entities, there should be a `water_heater.altherma`, click that. If not, please attach a screen shot and device diagnostics

**Comment by arn1s-m** [comment]
No, you can't set that in app either. Only on the unit itself

**Comment by McTrex** [comment]
That's a pity, what I'd want is to fire up the hot water tank heater when the energy prices are low or negative.
So, how come this is included in the other integration? Is that because it is only available through the LAN controller? Or do we have to wait until this feature becomes available in the Onecta app to be able to implement it?

**Comment by arn1s-m** [comment]
I don't see this available from the other integration either. Both are limited to actions available in the app.
I've personally tried to set Performance mode for such occasion. But it only works if you have <50 degree tank

**Comment by McTrex** [comment]
From my first post: Get/Set target temperature. And see the screenshot here:
https://github.com/tadasdanielius/daikin_altherma/blob/main/img/ha_altherma2.png

It shows an adjustable target temperature, that's the only thing missing here I think.

**Comment by jwillemsen** [comment]
It depends on the control mode of the water heater,  check `sensor.altherma_info_tank_setpoint_mode`. Can you control the target temperature of the tank using the Onecta app?

**Comment by McTrex** [comment]
No, unfortunately not. Maybe in a future software update...

**Comment by jwillemsen** [comment]
Please attach the diagnostics (json data), setting the target temperature of the tank could be weather dependent, at that moment you can't set it from onecta, you could change that on the altherma itself through the installer settings.

**Comment by McTrex** [comment]
[config_entry-daikin_residential_altherma-855ab489997eef71a57dc11ebaa05842.json.txt](https://github.com/speleolontra/daikin_residential_altherma/files/11704773/config_entry-daikin_residential_altherma-855ab489997eef71a57dc11ebaa05842.json.txt)

Please see attached the diagnostics, with some values edited for privacy.

**Comment by jwillemsen** [comment]
Ok, looking at the data, `setpointMode` for the tank is `false`, meaning we can't set it, so that feature is not enabled in HA. I do see that `operationMode` is set to `heating`, so not weather dependent. But, your `heatupMode` is set to `scheduleOnly` so meaning the reheat can only be controlled by your schedule. Try to disable the schedule, check if you can change the tank temperature in the onecta app at that moment, when you can change it there, reload this integration in your HA, at that moment you should be able to control the tank temperature also from HA

**Comment by McTrex** [comment]
Ok, in the installer menu I changed the heatupMode to fixed, now I can change the target temperature in the app between 30 and 60 degrees C.
When adding a thermostat card to HA, I get the error "Specify an entity from within the climate domain", see screenshot attached.
![Climate](https://github.com/speleolontra/daikin_residential_altherma/assets/29703173/412f4d64-8de5-4e40-b091-67aa517cb24b)

**Comment by jwillemsen** [comment]
The thermostat card is pure for climate entities, for the water heater devices you can use a regular entity card, or search for a custom card on the web. I use an entity card, haven't tried a custom card

**Comment by McTrex** [comment]
With the entity card I still see the target temperature. It also updates when I change it through the app. But no way to change it from HA, unfortunately. Would be nice to have.
Anyway, thanks for the quick responses and for your work on this integration!

**Comment by jwillemsen** [comment]
Try to restart HA

**Comment by McTrex** [comment]
It works! Thank you so much! Now I can trigger the heater whenever electricity prices are negative. Cool!
![Tanktemp](https://github.com/speleolontra/daikin_residential_altherma/assets/29703173/e508e324-ca16-49c8-bfca-4d06f63dc944)

**Comment by jwillemsen** [comment]
Keep an eye on the power usage, above 25c outsidr temperature the altherma uses the backup heater instead of the external unit, giving you a cop of 1 at that moment

**Comment by jwillemsen** [comment]
With negative prices heating with cop 1 is easy money.

**Comment by heikone** [comment]
Thanks for the answer!

Not sure, if we are talking about the same unit. I'm having a Daikin Altherma 3 R from 2022 and this one shall deliver 55°C hot water without boosting. Anyway, the behavior seems to be the same. I will report this also to Daikin and wait for reply. Depending on this reply, I will also contact my local heating installer, what to do, if the answer is not only "wait for coming update".

Thanks for the link to the Dutch forum, but because I'm not able to locate the exact post in all the 41 pages, could you please provide the date of the post, you are referring?

Good luck for Daikin support visit - had in sum more the 5 visits, because many things had been wrong after initial installation. From leaking water pipes, damaged valves, loose metal sheets and refrigerant leak!

**Comment by jwillemsen** [comment]
> Thanks for the link to the Dutch forum, but because I'm not able to locate the exact post in all the 41 pages, could you please provide the date of the post, you are referring?

This one specific https://gathering.tweakers.net/forum/list_message/75440092#75440092
